File: 004/Euler4.py
def palindrome(xs):
    j = str(xs)
    return (j == j[::-1])

def main():
    maximum = 0
    for h in xrange(100,1000):
        for i in xrange(100, 1000):
            j = h*i 
            if palindrome(j) and j > maximum:
                maximum = j
    return maximum

if __name__ == '__main__':
    print main()

File: 004/Euler4.rs
fn palindrome(x:uint) -> bool  {
    let st = str::to_chars(fmt!("%u", x));
    let rev = vec::reversed(st);
    rev == st
}

fn main() {
    let mut maximum = 0;
    for uint::range(100,1000) |x| {
        for uint::range(100, 1000) |y| {
            let j = x * y;
            if palindrome(j) && j > maximum{
                maximum = j;
            }
        }
        
    }
    io::println(fmt!("%u", maximum));
}